src/bundle/Resources/public/js/scripts/admin.limitation.pick.js

```diff
@@ -1,19 +1,164 @@
-    const selectLocationsConfirm = (target, data) => {
-        const selectedItems = data.reduce((total, item) => total + `<li>${item.ContentInfo.Content.Name}</li>`, '');
+    const udwContainer = doc.getElementById('react-udw');
+    const limitationBtns = doc.querySelectorAll(SELECTOR_LOCATION_LIMITATION_BTN);
+    const findLocationsByIdList = (pathArraysWithoutRoot, callback) => {
+        const bulkOperations = getBulkOperations(pathArraysWithoutRoot);
+        const request = new Request('/api/ezp/v2/bulk', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.BulkOperationResponse+json',
+                'Content-Type': 'application/vnd.ez.api.BulkOperation+json',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            body: JSON.stringify({
+                bulkOperations: {
+                    operations: bulkOperations,
+                },
+            }),
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
+        const errorMessage = Translator.trans(
+            /*@Desc("Failed to fetch content names")*/ 'limitation.pick.error',
+            {},
+            'universal_discovery_widget'
+        );
+
+        fetch(request)
+            .then(eZ.helpers.request.getJsonFromResponse)
+            .then(callback)
+            .catch(() => eZ.helpers.notification.showErrorNotification(errorMessage));
+    };
+    const getBulkOperations = (pathArraysWithoutRoot) =>
+        pathArraysWithoutRoot.reduce((operations, pathArray) => {
+            const locationId = pathArray[pathArray.length - 1];
+
+            operations[locationId] = {
+                uri: '/api/ezp/v2/views',
+                method: 'POST',
+                headers: {
+                    Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                    'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                    'X-Requested-With': 'XMLHttpRequest',
+                },
+                content: JSON.stringify({
+                    ViewInput: {
+                        identifier: `udw-locations-by-path-string-${pathArray.join('-')}`,
+                        public: false,
+                        LocationQuery: {
+                            Criteria: {},
+                            FacetBuilders: {},
+                            SortClauses: { SectionIdentifier: 'ascending' },
+                            Filter: { LocationIdCriterion: pathArray.join(IDS_SEPARATOR) },
+                            limit: 50,
+                            offset: 0,
+                        },
+                    },
+                }),
+            };
+
+            return operations;
+        }, {});
+    const removeRootLocation = (pathArray) => pathArray.slice(1);
+    const pathStringToPathArray = (pathString) => pathString.slice(1, -1).split('/');
```

Comment:
You should check whether `/` exists at the beginning and at the end of string. Otherwise, it can lead to unexpected results.